### PR TITLE
Add support for Github actions

### LINKED ISSUE #118: use github actions as ci/cd
[GitHub actions](http://github.com/features/actions) is a new workflow automation (~ CI/CD) service from GitHub, still in limited beta as of now.
I think it has some potential for R.

I've written a package for quickly getting started with GitHub actions: [ghactions](http://www.maxheld.de/ghactions/).

Would you guys be interested in supporting GitHub actions as a "backend" CI/CD provider?

If so, medium term, I'd like to work on supplying that backend through ghactions (much as the travis pkg currently does for tic).

It's obviously still very early days for github actions and the still maturing ghactions package, so it'll be some time before this makes sense.

Just wanted to raise this here to get your feedback and also to avoid doing duplicate work.

(also tracked in https://github.com/maxheld83/ghactions/issues/56)

### PR DESCRIPTION
**Pull request by pat-s**
Fixes #118 

## Changes

- New `tic::use_ghactions_deploy()` (status "experimental") to set up a SSH key for deployment.
  Needs [{sodium}](https://github.com/jeroen/sodium) for encryption of the secret being pushed via the API which again relies on the system lib `sodium`.
  This function also requires {gh} and {purrr} and is mostly c/p from {travis}.
  We might want to outsource it to a client package or think over bundling all within {tic}.

- New `use_ghactions_yml()` with `deploy = TRUE/FALSE` (FALSE by default).
  There will be a message after using this functions to comment in/out the runners a user wants to use by hand.
  Even though `use_ghactions_yml()` does not follow the structure of the other `use_*_yml()` functions I did not want to make the fun more complicated than it needs to be.

- Proposed change for the name of the private deploy key from `TRAVIS_DEPLOY_KEY` to `TIC_DEPLOY_KEY` to have a generic name.
  I still dislike `id_rsa` and think that a descriptive name with tic in its name is more descriptive, especially as it has to be added as an env var in the GH actions builds.

- Inline with the comment above: Proposed change from `travis_private_key_name` to `private_key_name`

- New Vignette "FAQ".

- Added GH Actions to `use_tic()` wizard.

- Renamed `yaml-templates.R` to `yaml_templates.R` because the former caused troubles when previewing the dev version of the docs.

- Beautified the CLI output of `use_tic()`

- Replaced all instances of `_tic_` in the doc by `{tic}`

- Honor env var `pkgType` if set to set `options(pkgType="source")` so that builds do not fail on macOS-devel where no binaries are available. [Example errored build](https://github.com/pat-s/oddsratio/runs/454659747?check_suite_focus=true#step:14:351) without `pkgType`.

- use `RGL_USE_NULL: true` to prevent issues with X11 display for {rgl}

- rename arg `name` to `private_key_name`

- use [CRANs custom `clang` builds](https://cran.r-project.org/bin/macosx/tools/) on macOS

- Got rid of all warnings caused by `ssh`. These trigger "annotations" in the GH Actions summary. Now running `ssh` commands with `LogLevel=error`

## Example

See [tic.package](https://github.com/ropenscilabs/tic.package/actions) for a practical example of this branch in action.

### CODE REVIEW

File: R/macro-bookdown.R [hunk 1]
```diff
@@ -46,9 +41,7 @@ do_bookdown <- function(...,
                         remote_url = NULL,
                         commit_message = NULL,
                         commit_paths = ".",
-                        travis_private_key_name = "TRAVIS_DEPLOY_KEY") {
-
-  name <- travis_private_key_name
```
Comment (krlmlr) [inline R/macro-bookdown.R]: If we touch this, perhaps get rid of `name` and use `private_key_name`?

File: R/macro-bookdown.R [hunk 2]
```diff
@@ -84,7 +79,7 @@ do_bookdown <- function(...,
     #' 1. [step_setup_push_deploy()] in the `"before_deploy"` stage
     #'    (if `deploy` is set),
     #'
-    name <- travis_private_key_name
+    name <- private_key_name
     get_stage("before_deploy") %>%
       add_step(step_setup_ssh(name = name)) %>%
```
Comment (krlmlr) [inline R/macro-bookdown.R]:
```suggestion
      add_step(step_setup_ssh(name = !!private_key_name)) %>%
```

File: R/macro-pkgdown.R [hunk 3]
```diff
@@ -81,7 +76,7 @@ do_pkgdown <- function(...,
   if (isTRUE(deploy)) {
     #' 1. [step_setup_ssh()] in the `"before_deploy"` to setup
     #'    the upcoming deployment (if `deploy` is set and only on Travis CI),
-    if (ci_on_travis()) {
+    if (ci_on_travis() | ci_on_ghactions()) {
```
Comment (krlmlr) [inline R/macro-pkgdown.R]:
```suggestion
    if (ci_on_travis() || ci_on_ghactions()) {
```

File: R/macro-pkgdown.R [hunk 4]
```diff
@@ -47,9 +43,7 @@ do_pkgdown <- function(...,
                        remote_url = NULL,
                        commit_message = NULL,
                        commit_paths = ".",
-                       travis_private_key_name = "TRAVIS_DEPLOY_KEY") {
-
-  name <- travis_private_key_name
```
Comment (krlmlr) [inline R/macro-pkgdown.R]: Same.

File: R/steps-install.R [hunk 5]
```diff
@@ -14,12 +14,17 @@ InstallDeps <- R6Class(
     },
 
     run = function() {
+      if (Sys.getenv("pkgType") != "") {
+        private$type <- Sys.getenv("pkgType")
+        crayon::magenta(cli_text("Using '{private$type}' for argument {.code pkgType} in {.code install.packages()}."))
```
Comment (krlmlr) [inline R/steps-install.R]:
```suggestion
        crayon::magenta(cli_text('Using "{private$type}" for argument {.code pkgType} in {.code install.packages()}.'))
```

File: R/steps-install.R [hunk 6]
```diff
@@ -81,10 +86,17 @@ InstallCRAN <- R6Class(
       private$install_args <- list(...)
     },
     run = function() {
+      if (Sys.getenv("pkgType") != "") {
+        pkgType <- Sys.getenv("pkgType")
+        cli_text("Using '{pkgType}' for argument {.code pkgType} in {.code install.packages()}.")
```
Comment (krlmlr) [inline R/steps-install.R]: Same. I think it's worth extracting a function here.

File: R/steps-install.R [hunk 7]
```diff
@@ -81,10 +86,17 @@ InstallCRAN <- R6Class(
...
+      }
       if (length(find.package(private$package, quiet = TRUE)) == 0) {
-        do.call(
-          install.packages,
-          c(list(pkg = private$package), private$install_args)
+        withr::with_options(
+          c(pkgType = pkgType),
+          do.call(
```
Comment (krlmlr) [inline R/steps-install.R]: Can we use `rlang::exec()`?

File: R/steps-rcmdcheck.R [hunk 8]
```diff
@@ -153,11 +153,12 @@ step_rcmdcheck <- function(...,
   #' @param args `[character]`\cr
   #'   Passed to `rcmdcheck::rcmdcheck()`.\cr
   #'   Default for Travis and local runs: `c("--no-manual", "--as-cran")`.\cr
-  #'   Default for Appveyor:
+  #'   Default for Appveyor and GH Actions (Windows):
   #'   `c("--no-manual", "--as-cran", "--no-vignettes",
   #'   "--no-build-vignettes", "--no-multiarch")`.\cr
   if (is.null(args)) {
-    if (isTRUE(ci_on_appveyor())) {
+    if (isTRUE(ci_on_appveyor()) |
```
Comment (krlmlr) [inline R/steps-rcmdcheck.R]: `||` implements a shortcut, the RHS is not evaluated if the LHS is true.

```suggestion
    if (isTRUE(ci_on_appveyor()) ||
```

File: R/steps-ssh.R [hunk 9]
```diff
@@ -136,15 +136,16 @@ InstallSSHKeys <- R6Class(
 #' to a file in `~/.ssh`.
 #' Only run in non-interactive settings and if the environment variable
 #' exists and is non-empty.
-#' The [travis::use_travis_deploy()] and [use_tic()] functions encode a private
-#' key as an environment variable for use with this function.
+#' [travis::use_travis_deploy()] , [use_ghactions_deploy()] and[use_tic()]
```
Comment (krlmlr) [inline R/steps-ssh.R]:
```suggestion
#' [travis::use_travis_deploy()] , [use_ghactions_deploy()] and [use_tic()]
```

File: R/use_tic.R [hunk 10]
```diff
@@ -346,6 +371,7 @@ ci_menu <- function(choices, title) {
     travis = "Travis CI",
     circle = "Circle CI",
     appveyor = "AppVeyor CI",
+    ghactions = "Github Actions",
```
Comment (krlmlr) [inline R/use_tic.R]: Should we be consistent with the trademark here?

```suggestion
    ghactions = "GitHub Actions",
```

File: codemeta.json [hunk 11]
```diff
@@ -402,7 +450,7 @@
   ],
   "releaseNotes": "https://github.com/ropensci/tic/blob/master/NEWS.md",
   "readme": "https://github.com/ropensci/tic/blob/master/README.md",
-  "fileSize": "350.164KB",
+  "fileSize": "14375.312KB",
```
Comment (krlmlr) [inline codemeta.json]: That's a lot of stuff.

File: R/steps-rcmdcheck.R (hunk 8 above)
Comment (pat-s) [reply R/steps-rcmdcheck.R]: I'm aware of the shortcut - but you are right, `||` is correct here :)

File: R/use_tic.R (hunk 10 above)
Comment (pat-s) [reply R/use_tic.R]: Ok, then I'll do it pkg wide.

File: codemeta.json (hunk 11 above)
Comment (pat-s) [reply codemeta.json]: Indeed - wondering why. Didn't check the number before.

Comment (pat-s) [reply codemeta.json]: Found it - for some reason I still had a `.drake` folder in the repo.

**Comment by pat-s** [comment]
> Note that {ghactions} has been superseded by a repo of workflow files: r-lib/actions.

I know. Talked to @maxheld83 already and we will join forces - probably {ghactions} will be transfered to ropensci and join the CI related packages group; templating will move into {tic}.

**Comment by pat-s** [comment]
- [ ] Use clang7 for macOS instead of `/usr/bin/clang` because we need an openMP enabled compiler to compile packages on R-devel successfully

**Comment by pat-s** [comment]
- Got rid of all warnings caused by `ssh`. These trigger "annotations" in the GH Actions summary. Now running `ssh` commands with `LogLevel=error`

**Comment by pat-s** [comment]
I'd say I'm happy now with the current state.

Testing it in two repos since a few days:

- https://github.com/ropenscilabs/tic.package/actions
- https://github.com/mlr-org/mlr3filters/actions

Problematic pkgs like {rgl} and {rJava} should work on any OS. 
Feel free to give it a shot in one of your repos.

```r
tic::use_ghactions_deploy()
tic::use_ghactions_yml(deploy = TRUE)

# tic::use_tic_badge("ghactions") # from #230 
```

**Comment by pat-s** [comment]
@krlmlr I am merging this now to open the "testing" to the public via the master branch.
Feel free to come back here and comment on the PR - I'll integrate changes in smaller commits along the way then :)

**Comment by krlmlr** [comment]
Sounds good. I'll test very soon.

**Pull request merged**